subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java

```diff
@@ -289,7 +289,7 @@ private GroovyJavaJointCompileSpec createSpec() {
 
         spec.setSourcesRoots(sourceRoots);
         spec.setSourceFiles(stableSourcesAsFileTree);
-        spec.setDestinationDir(getDestinationDir());
+        spec.setDestinationDir(getDestinationDirectory().getAsFile().getOrNull());
```

Comment:
This should call `get()` instead of `getOrNull()`, as the destination directory should not be null.